logging/logging_test.go

```diff
@@ -609,6 +612,132 @@ func TestToLogEntry(t *testing.T) {
 	}
 }
 
+func TestToLogEntryOTelIntegration(t *testing.T) {
+	// Some slight modifications need to be done for testing ToLogEntry
+	// for the OpenTelemetry integration, so they are in a separate function.
+	u := &url.URL{Scheme: "http"}
+	tests := []struct {
+		name string
+		in   logging.Entry
+		want *logpb.LogEntry // if want is nil, pull wants from spanContext
+	}{
+		{
+			name: "Using OpenTelemetry with a valid span",
+			in: logging.Entry{
+				HTTPRequest: &logging.HTTPRequest{
+					Request: &http.Request{
+						URL: u,
+					},
+				},
+			},
+		},
+		{
+			name: "Using OpenTelemetry only with a valid span + valid traceparent headers (precedence test)",
+			in: logging.Entry{
+				HTTPRequest: &logging.HTTPRequest{
+					Request: &http.Request{
+						URL: u,
+						Header: http.Header{
+							"Traceparent": {"00-105445aa7843bc8bf206b12000100012-000000000000004a-01"},
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "Using OpenTelemetry only with a valid span + valid XCTC headers (precedence test)",
+			in: logging.Entry{
+				HTTPRequest: &logging.HTTPRequest{
+					Request: &http.Request{
+						URL: u,
+						Header: http.Header{
+							"X-Cloud-Trace-Context": {"105445aa7843bc8bf206b120000000/0000000000000bbb;o=1"},
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "Using OpenTelemetry with a valid span + trace info set in Entry object",
+			in: logging.Entry{
+				HTTPRequest: &logging.HTTPRequest{
+					Request: &http.Request{
+						URL: u,
+					},
+				},
+				Trace:        "abc",
+				SpanID:       "def",
+				TraceSampled: false,
+			},
+			want: &logpb.LogEntry{
+				Trace:        "abc",
+				SpanId:       "def",
+				TraceSampled: false,
+			},
+		},
+		{
+			name: "Using OpenTelemetry without a request",
+			in:   logging.Entry{},
+			want: &logpb.LogEntry{},
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			var span trace.Span
+			ctx := context.Background()
+
+			// Set up an OTel SDK tracer if integration test, mock noop tracer if not.
+			if integrationTest {
```

Comment:
Do we have a dedicated places for integration tests for Go Logging library?
For example, for Java-logging and Node-logging, there is a separate folder for integration tests:
https://github.com/googleapis/java-logging/tree/main/google-cloud-logging/src/test/java/com/google/cloud/logging/it
https://github.com/googleapis/nodejs-logging/tree/main/system-test

Reply:
I don't think we have any right now, but if enough system tests happen that are kinda different from integration tests controlled by the `integrationTest` flag I could make a separate file for them.